sktime/forecasting/compose/_pipeline.py

```diff
@@ -1462,6 +1488,23 @@ def _get_forecaster_X_prediction(self, X=None, fh=None, method="predict"):
             return None
         if isinstance(self.columns, (list, pd.Index)) and len(self.columns) == 0:
             return X
+
+        # if user passes data for future unknown variables, do not forecast them
+        # this is done only if predict_behaviour is "use_actuals"
+        if isinstance(self.columns, (list, pd.Index)) and len(self.columns):
+            is_future_unknown_actually_known = all(
+                column in X.columns for column in self.columns
+            )
+        elif X is None:
+            is_future_unknown_actually_known = False
```

Comment:
If `predict` received no `X` argument, of course everything is missing. The check fails and we must go to `forecaster_X`.

Reply:
I probably should have made this as the first check instead of as second.